### PR TITLE
Fixing a bug in tf.parallel_stack

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**
This PR fixes the bug reported in this issue #27521 .
It also updates the unit test to add one more test case since this bug only shows up with large tensors.

### CODE REVIEW

**Review by shahzadlone — APPROVED** [review]

File: tensorflow/core/common_runtime/parallel_concat_optimizer.cc [hunk 1]
```diff
@@ -89,13 +88,11 @@ class ParallelConcatRemovePass : public GraphOptimizationPass {
         Node* update;
         TF_RETURN_IF_ERROR(
             make_node("_ParallelConcatUpdate")
-                .Attr("loc", i)
+                .Attr("loc", input_edge->dst_input())
```
Comment (rmlarsen) [inline tensorflow/core/common_runtime/parallel_concat_optimizer.cc]: Sigh.

**Review by rmlarsen — APPROVED** [review]
Thank you for tracking this down!

**Review by alextp — APPROVED** [review]

**Comment by mahmoud-abuzaina** [comment]
@rthadur it looks like the failures here are not related to the PR.

**Comment by mahmoud-abuzaina** [comment]
@rthadur can you let us know what internal test (feedback/copybara) is failing?

**Comment by alextp** [comment]
@mahmoud-abuzaina it looks like unrelated flakes

**Comment by rmlarsen** [comment]
Woohoo!

**Pull request merged**